src/node.cc

```diff
@@ -205,7 +205,17 @@ void Environment::InitializeInspector(
     return;
   }
 
+  if (should_wait_for_inspector_frontend()) {
+    WaitForInspectorFrontend();
+  }
+
   profiler::StartProfilers(this);
+}
+
+void Environment::WaitForInspectorFrontend() {
+  if (!inspector_agent_->WaitForConnectByOptions()) {
+    return;
+  }
 
   if (inspector_agent_->options().break_node_first_line) {
     inspector_agent_->PauseOnNextJavascriptStatement("Break at bootstrap");
```

Comment:
When I first looked at the function name, I expected it to only wait for the frontend connection. But, it also sets a breakpoint at the first line if specified in the options. Does it make sense for this function to handle setting the breakpoint as well?

Reply:
The function name omitted words like 'based on options set at creation,' which I initially thought was okay. I renamed it to make it clearer. PTAL.